### PR TITLE
RDKShell: Skip SetFocus to false for Hibernated plugins

### PR DESCRIPTION
**Pull request by adrianM27**

### CODE REVIEW

File: RDKShell/RDKShell.cpp [hunk 1]
```diff
@@ -6606,13 +6606,39 @@ namespace WPEFramework {
...
+                            //Check plugin state, skip for hibernated state
+                            string method = "status@" + previousFocusedIterator->first;
+                            Core::JSON::ArrayType<PluginHost::MetaData::Service> joResult;
+                            auto thunderController = getThunderControllerClient();
+                            uint32_t ret = thunderController->Get<Core::JSON::ArrayType<PluginHost::MetaData::Service>>(RDKSHELL_THUNDER_TIMEOUT, method.c_str(), joResult);
+                            bool isHibernated = false;
+                            if (ret == Core::ERROR_NONE && joResult.Length() > 0)
+                            {
+                                PluginHost::MetaData::Service service = joResult[0];
+                                if (service.JSONState == PluginHost::MetaData::Service::state::HIBERNATED)
```
Comment (adrianM27) [inline RDKShell/RDKShell.cpp:6619]: worth to consider excluding SUSPENDED state, only If we confirm the call to Focused(false) is not needed for hibernated or suspended apps

**Comment by goruklu** [comment]
this looks good to me but @binuinbaraj should review as well.

File: RDKShell/RDKShell.cpp [hunk 2]
```diff
@@ -3835,6 +3851,12 @@ namespace WPEFramework {
                     }
                 }
 
+                //Reset app suspended/hibernated for launch
```
Comment (goruklu) [inline RDKShell/RDKShell.cpp:3854]: Might be good to use the #ifdef HIBERNATE_SUPPORT_ENABLED everywhere we use suspendOrHibernateApps array and mutex.

File: RDKShell/RDKShell.cpp [hunk 3]
```diff
@@ -197,6 +197,9 @@ bool sForceResidentAppLaunch = false;
 static bool sRunning = true;
 bool needsScreenshot = false;
 
+std::mutex suspendedOrHibernatedAppsMutex;
```
Comment (goruklu) [inline RDKShell/RDKShell.cpp:200]: Might be good to use the #ifdef HIBERNATE_SUPPORT_ENABLED everywhere we use suspendOrHibernateApps array and mutex.

File: RDKShell/RDKShell.cpp [hunk 4]
```diff
@@ -606,6 +609,10 @@ namespace WPEFramework {
                mLaunchEnabled = false;
             }
 
+            suspendedOrHibernatedAppsMutex.lock();
```
Comment (goruklu) [inline RDKShell/RDKShell.cpp:612]: Might be good to use the #ifdef HIBERNATE_SUPPORT_ENABLED everywhere we use suspendOrHibernateApps array and mutex.

**Review by goruklu — CHANGES_REQUESTED** [review]

**Review by binuinbaraj — COMMENTED** [review]
Changes looks good.

**Review by goruklu — APPROVED** [review]

**Pull request merged**